### PR TITLE
[CCXDEV-11814] Add payload tracker

### PR DESCRIPTION
**Pull request by epapbak**
# Description

- Add Kafka producer
- Add payload tracker (uses Kafka producer) and track status of message processing
- Add corresponding UTs
- Remove types duplicated in [insights-results-types](https://github.com/RedHatInsights/insights-results-types/blob/master/types.go)

Fixes CCXDEV-11814

## Type of change

- New feature (non-breaking change which adds functionality)

## Testing steps

CI

## Checklist
* [x] `make before_commit` passes
* [x] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: kafka_producer.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+// sent to the Payload Tracker topic in Kafka.
+type PayloadTrackerMessage struct {
+	Service   string `json:"service"`
+	RequestID string `json:"request_id"`
+	Status    string `json:"status"`
+	Date      string `json:"date"`
+}
+
+// NewProducer instantiates a Kafka Producer
+func NewProducer(config *BrokerConfiguration) (*Producer, error) {
```
Comment (Bee-lee) [inline kafka_producer.go:71]: do we not have something like this in other repos?

**Review by Bee-lee — APPROVED** [review]
LGTM overall

File: kafka_producer.go (hunk 1 above)
Comment (epapbak) [reply kafka_producer.go:71]: we do, and I started the PR by refactoring and putting common code together in io-utils and ran into an issue with dependencies. So I decided to leave the refactor for another time, again 😞

**Review by JiriPapousek — APPROVED** [review]

File: kafka_producer.go (hunk 1 above)
Comment (juandspy) [reply kafka_producer.go:71]: shall we create a task to unify producers in our Go code (?) this way this PR won't be blocked but we will eventually reduce code duplicity

**Review by juandspy — APPROVED** [review]

**Pull request merged**